### PR TITLE
AF-2691: Open projects directly using the URL

### PR DESCRIPTION
**Pull request by paulovmr**
**JIRA**: [AF-2691](https://issues.redhat.com/browse/AF-2691)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins do fdb

**Comment by paulovmr** [comment]
jenkins do fdb

**Comment by tomasdavidorg** [comment]
Hi @paulovmr , I will take a look on this PR tomorrow.

**Comment by paulovmr** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/perspective/LibraryPerspective.java [hunk 1]
```diff
@@ -66,18 +91,33 @@ public PerspectiveDefinition buildPerspective() {
 
     @OnStartup
     public void onStartup(final PlaceRequest placeRequest) {
+        this.projectPath = (((getWindowParameterMap().containsKey("path")) ? getWindowParameterMap().get("path").get(0) : "")).trim();
```
Comment (tomasdavidorg) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/perspective/LibraryPerspective.java:94]:
```suggestion
        this.projectPath = (getWindowParameterMap().containsKey("path") ? getWindowParameterMap().get("path").get(0) : "").trim();
```

I think the parentheses are not necessary.

**Review by tomasdavidorg — APPROVED** [review]

**Comment by paulovmr** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/screens/importrepository/ImportRepositoryPopUpPresenterTest.java [hunk 2]
```diff
@@ -174,7 +175,7 @@ public void importEmptyRepositoryOnSuccessTest() {
         verify(view).hideBusyIndicator();
         verify(view).hide();
         verify(notificationEvent).fire(any(NotificationEvent.class));
-        verify(libraryPlaces).goToProject(any());
+        verify(libraryPlaces).goToProject(any(WorkspaceProject.class));
```
Comment (caponetto) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/screens/importrepository/ImportRepositoryPopUpPresenterTest.java:178]: Is this change really necessary? If so, I'll have to update this part on our JDK11 PR.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/screens/importrepository/ImportRepositoryPopUpPresenterTest.java [hunk 3]
```diff
@@ -195,7 +196,7 @@ public void importEmptyRepositoryOnFailureTest() {
         verify(view).hideBusyIndicator();
         verify(view, never()).hide();
         verify(notificationEvent, never()).fire(any(NotificationEvent.class));
-        verify(libraryPlaces, never()).goToProject(any());
+        verify(libraryPlaces, never()).goToProject(any(WorkspaceProject.class));
```
Comment (caponetto) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/test/java/org/kie/workbench/common/screens/library/client/screens/importrepository/ImportRepositoryPopUpPresenterTest.java:199]: Is this change really necessary? If so, I'll have to update this part on our JDK11 PR.

**Review by caponetto — APPROVED** [review]
LGTM @paulovmr! I just left a question.

**Pull request merged**